### PR TITLE
WIP feat(tabs): add experimental tabs component 

### LINKED ISSUE #1293: Experimental Component - Tabs - Development
**Development**
A developer can discover and install the experimental component by following accessible documentation.

- [ ] Add to carbon-components
- [ ] Update website (add to the experimental section and update component status page)

### PR DESCRIPTION
**Pull request by jnm2377**
Closes #1293 
Adds experimental tabs component.

#### Changelog

**New**
- adds  experimental tabs styling to `_tabs.scss`
- adds experimental tab styling variables to `_theme-tokens.scss` 
- adds new class `.#{$prefix}--tabs-trigger--open` for arrow motion on open/close in dropdown view
- adds testing for disabled tab and new `.#{$prefix}--tabs-trigger--open` class
- adds new `.#{$prefix}--tabs-trigger--open` class to default tab options in `tabs.js`
- adds functionality to include `disabled` in html `li` element if tab is disabled
- adds `@mixin` to `_layout.scss` to use for mobile media queries (allows for *max-width*, in this case used particularly for styling the tab dropdown in mobile view)
- adds `:active` state on links for Safari focus states

**Changed**
- refactor `_tabs.scss` file to use `@mixin`
- refactor scss code for experimental tabs to no longer be nested (except for `@include breakpoint()`'s.
- refactor `tabs.config.js` to include a disabled tab item
- refactor `tabs.js` to include functionality for disabled tab
- refactor `_updateMenuState(force)` to toggle new `.#{$prefix}--tabs-trigger--open`

**Testing/Reviewing**
- Ensure *disabled* styling is functioning correctly on hover, focus, select on click and keyboard navigation.
- Ensure mobile view is functioning correctly.
- Ensure focus states in Safari are working.
- Staging link: http://carbon-dev-environment-tabs-x-grateful-camel.mybluemix.net/component/tabs--default.html

### CODE REVIEW

**Comment by alisonjoseph** [comment]
Tagging @jeanservaas -- see above question about responsive tabs? 🤔

**Comment by tw15egan** [comment]
Should it just mimic the default dropdown styles in mobile?

File: src/components/tabs/tabs.hbs [hunk 1]
```diff
@@ -11,6 +11,9 @@
         <a id="{{linkId}}" class="bx--tabs__nav-link" href="javascript:void(0)" role="tab" aria-controls="{{panelId}}"{{#if selected}} aria-selected="true"{{/if}}>{{label}}</a>
       </li>
     {{/each}}
+      <li class="bx--tabs__nav-item" data-target=".tab-6" role="presentation" disabled>
```
Comment (asudoh) [inline src/components/tabs/tabs.hbs]: Wondering if we can merge here to the `{{#each}}` loop? We may add `{{#if disabled}} disabled{{/if}}` and update the `tabs.config.js` accordingly.

**Review by asudoh — COMMENTED** [review]
LGTM 👍 basically, just one thing - Thanks @jnm2377!

**Comment by alisonjoseph** [comment]
@jnm2377 I added a `WIP` label, when you're ready for review again you can remove it 😸

File: src/components/tabs/tabs.hbs (hunk 1 above)
Comment (jnm2377) [reply src/components/tabs/tabs.hbs]: @asudoh great suggestion! I went ahead and did that. 😄

**Comment by laurenmrice** [comment]
Some comments on the tabs that convert to dropdown @jnm2377 :


-The horizontal lines between tab options need to start 16 px from each side of the background. (see image)


-Transitions are different in this dropdown than transitions on the experimental dropdown, should they be the same?

-Type is slightly too high in the field and needs to be more vertically center.

-The selection at the top of the dropdown should have a background Gray 20 when the drawer is open and should still have that background when you are hovering on items below it.

-Relating to above comment, the focus is activated once you have clicked the dropdown to open it and is still active when you are hovering over other items. The focus should go away until you make a selection.

-The color of the text inside the drawer should be Gray 70 instead of Gray 100. The text will turn Gray 100 on hover. This is a fix we also need to make eventually in the dropdown experimental component. :)

**Comment by jeanservaas** [comment]
**Dropdown component:**

Since most of these questions are about the dropdown pattern, def. refer to @laurenmrice 's comments on dropdown above, that was her file so she knows it inside out.

Also talked to @aagonzales about these questions: 

_1. Transitions on open:
 I noticed that the standard tabs dropdown has no transitions on open..._

_2. Selected on open:
 Triggerbox issue... dropdown takes selection out of list... tabs component doesn't..._

_3. Hover on open:
Experimental dropdown component highlights 2 items on hover: any item you are hovering on in the menu and the trigger box item even if you're not specifically hovering over the trigger box. Standard tabs do not highlight trigger box..._

To the points above, the way the Tabs are functioning seems like a glitch in Carbon... When Tabs are collapsed, they should function as a true dropdown... and dropdowns should function the same way everywhere... I know we're supposed to be mapping to current Carbon functionality, so I hope this isn't a problem to change, but everyone seems to agree it's an inconsistency. 

Also, I think @tw15egan made an update to the dropdown component (it's still in a PR somewhere) where the selected trigger box isn't a hover... it's a selected state and stays Gray 20 (see structured list component) when the dropdown is open.

**As for the tabs component:**

- Can we fix the padding between the tabs when the focus state triggers... just so it maintains the padding rather than covers it? That should fix the overlap issue on the right side of the tab too, where the margin bottom is overlapping the focus box

screen shot 2018-11-12 at 9 49 19 am

Also, the text below the tab is weird... it's 16px while the tabs are 14px which looks a little odd and then the right alignment jumps when the browser scales... I know it's just placeholder stuff, but I'm wondering if we can keep the alignment consistent to the type and change sample text to 14px.

image

image

File: src/components/tabs/tabs.js [hunk 2]
```diff
@@ -168,10 +170,12 @@ class Tab extends ContentSwitcher {
       selectorTrigger: `.${prefix}--tabs-trigger`,
       selectorTriggerText: `.${prefix}--tabs-trigger-text`,
       selectorButton: `.${prefix}--tabs__nav-item`,
+      selectorButtonEnabled: `.${prefix}--tabs__nav-item:not([disabled])`,
       selectorButtonSelected: `.${prefix}--tabs__nav-item--selected`,
       selectorLink: `.${prefix}--tabs__nav-link`,
       classActive: `${prefix}--tabs__nav-item--selected`,
       classHidden: `${prefix}--tabs__nav--hidden`,
+      classOpen: `${prefix}--tabs-trigger--open`,
```
Comment (asudoh) [inline src/components/tabs/tabs.js:178]: Seems to reflect this to the test, as we did for `selectorButtonEnabled`.

**Review by asudoh — COMMENTED** [review]
Nice progress @jnm2377!

**Review by jeanservaas — APPROVED** [review]
Looks great in Chrome, all changes made, just an fyi, doesn't look like focus states are triggering in safari... you guys are probably aware of that.

File: src/components/tabs/tabs.js (hunk 2 above)
Comment (jnm2377) [reply src/components/tabs/tabs.js:178]: @asudoh you were right. I added testing for the new class and it passed now. Thank you! 😄

**Review by asudoh — APPROVED** [review]
LGTM 👍🎉 - Thanks @jnm2377!

**Pull request merged**